userdetails-plugin/grails-app/views/registration/createAccount.gsp

```diff
@@ -422,6 +446,110 @@
+                         if(result.includes(error)){
+                             document.getElementById("newEmailMessage").innerHTML = error
+                         }
+                         else {
+                            document.getElementById("newEmailMessage").innerHTML = 'Error please contact us at <a href="mailto:support@ala.org.au">support@ala.org.au</a>'
+                         }
+                         document.getElementById("newEmailMessage").style.color = "red"
+                         document.getElementById("newEmailMessage").hidden = false
+                         document.getElementById("emailCodeDiv").hidden = true
+                    }
+                }});
+             }
+         });
+
+         $("#verifyCode").click(function(e) {
+            var emailCode = $("#emailCode").val();
+             if(emailCode == null || emailCode === "") {
+                 document.getElementById("newEmailMessage").innerHTML = "Invalid code"
+                 document.getElementById("newEmailMessage").style.color = "red"
+                 document.getElementById("newEmailMessage").hidden = false
+                 document.getElementById("emailCodeDiv").hidden = false
+             }
+             else {
+                 $.ajax({
+                 url: "${createLink(uri:'/registration/verifyAttributeChangeWithCode')}?attribute=email&code=" + emailCode,
+                 type: "GET",
+                 success: function(result){
+                     if(result.success){
+                        window.location = "${createLink(uri:'/logout')}?url=/"
+                    }
+                     else{
+                         document.getElementById("newEmailMessage").innerHTML = result.error
+                         document.getElementById("newEmailMessage").style.color = "red"
+                         document.getElementById("newEmailMessage").hidden = false
+                         document.getElementById("emailCodeDiv").hidden = false
+                     }
+                }});
+             }
+         });
+
+         $("#updateEmail").click(function(e) {
+             var newEmail = $("#newEmail").val();
+             if(newEmail == null || newEmail === "") {
+                 document.getElementById("newEmailMessage").innerHTML = "Invalid email"
+                 document.getElementById("newEmailMessage").style.color = "red"
+                 document.getElementById("newEmailMessage").hidden = false
+             }
+             else {
+                 $.ajax({
+                 url: "${createLink(uri:'/registration/update')}?email=" + newEmail,
+                 type: "GET",
+                 success: function(result){
+                     let error = "A user is already registered with the email address"
+                     if(result.success) {
+                         $.ajax({
+                         url: "${createLink(uri:'/registration/deactivateAccountAndSendActivationEmail')}?email=" + newEmail,
```

Comment:
Encode new email

Reply:
Also, sending 2 requests for this seems wrong - the user could update their email then block the deactivateAccountAndSendActivationEmail call?  Surely the deactivateAccountAndSendUserEmail should happen atomically on the server side with the update if required?